Searchbar testing

voltrare opened the pull request:
update search bar for `Extensions`, `Sources` and `SourceLibrary`
inspired by search bar on tachiyomi

![image](https://user-images.githubusercontent.com/30526233/135480218-9830a928-0d8d-4b9e-ba6d-70c9ec7f5fb3.png)

voltrare committed:
update search bar for `Extensions` and `Sources`

voltrare committed:
update search bar for `SourceLibrary`

mgn-norm commented on src/qml/Extensions.qml:
```diff
@@ -44,8 +44,15 @@ Rectangle {
   TextField {
     id: searchBox
-    placeholderText: qsTr("Search")
+    placeholderText: qsTr("   Search")
```
instead of putting spaces before the text, you should add anchors and add a left margin to it. probably align it with the icons. this applies to all the other text fields as well.

voltrare replied:
ah alright

voltrare committed:
update `Sources`, `SourceLibrary` and` Extension` with icons

using `RowLayout` to bring icons and searchbar but used `Rectangle` to color the background for `SourceLibrary` searchbar
